src/storage/txn/commands/prewrite.rs

```diff
@@ -587,27 +592,32 @@ impl PrewriteKind for Optimistic {
     }
 }
 
+/// Information needed for the pessimistic variant of `PreWriteKind`.
 struct Pessimistic {
     for_update_ts: TimeStamp,
 }
 
+/// The pessimistic variant of `PreWriteKind`.
 impl PrewriteKind for Pessimistic {
+    // see the `MutationLock` trait implementation for `(Mutation, bool)`
     type Mutation = (Mutation, bool);
 
     fn txn_kind(&self) -> TransactionKind {
         TransactionKind::Pessimistic(self.for_update_ts)
     }
 }
 
-/// See PrewriteKind::Mutation. This is either just a mutation (for optimistic
-/// transactions, since they never include locks) or a mutation and a bool (for
-/// pessimistic transactions, where the bool indicates if the mutation is taking
-/// a pessimistic lock).
+/// Used as an associated type within `PrewriteKind::Mutation` to infer the kind of
+/// prewrite (optimistic/pessimistic).
```

Comment:
I think the kind of prewrite is not inferred from this associated type but instead determined by the kind of prewrite. So I suggest reusing the comment of `PrewriteKind::Mutation`.

Reply:
Done.